scripts/archive_links.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+# ruff: noqa: T201
+
+import argparse
+import json
+import os
+import re
+import sys
+import time
+from typing import Optional
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+EXTENSIONS_TO_SCAN = [".py", ".po"]
+FIND_URL_PATTERN = re.compile(r'https?://[^\s<>"]+|www\.[^\s<>"]+')
```

Comment:
In fact, we have no links that don't start with http(s), do we can simplify this regex and no check with `url.startswith` below.